### PR TITLE
Update grid settings to match prod

### PR DESCRIPTION
**Pull request by barneyfoxuk**
# Description

This PR originally was to update the grid maths to align more with how our previous grids have worked.  In an effort to please typescript I have ended up restructuring pretty much all of it though as we had mixed types being passed around that caused some fundamental issues. I think on the plus side the code is more readable now though?

# Checklist

Pull request contains:

- [ ] A new component
- [x] Component maintenance: improvement / bug fix / etc
- [ ] Component library change: storybook / webpack / etc

Definition of done:

- [ ] Includes theme changes for both Uswitch and money
- [ ] Work has been tested in multiple browsers
- [ ] PR description includes description of change
- [ ] PR description includes screenshot of change
- [ ] If new component, designer has approved screenshot
- [ ] If the change will affect other teams, that team knows about this change
- [ ] If introducing a new component behaviour, added a story to cover that case.
- [ ] If component change, version updated

### CODE REVIEW

File: src/layout/flex-grid/src/index.tsx [hunk 1]
```diff
@@ -107,7 +82,13 @@ export const Row: React.FC<RowProps> = ({
       sx={{
         variant: `grid.row`,
         mx: (theme: any): any =>
-          getNegativeSpaceValue(getGutterSize(theme))(theme),
+          removeArrayIfSingleItem(
+            alwaysGetArray(getGutterSize(theme))
+              .map((value: any) => getValueFromSize(theme, value))
```
Comment (callumacrae) [inline src/layout/flex-grid/src/index.tsx:86]: I don't think you need these types here as typescript should get them from the function signatures, right?

(not that it hurts or really makes any difference at all)

File: src/layout/flex-grid/src/index.tsx [hunk 2]
```diff
@@ -50,14 +12,23 @@ const getVerticalGutterSize = (theme: any): string | number =>
...
-  return fn(data, 0)
+const alwaysGetArray = (valueOrArray: any): any => [].concat(valueOrArray)
+
+const getValueFromSize = (theme: any, size: string | number): number =>
+  typeof size === 'number' ? size : theme.space[size] || 0
+
+const getHalf = (value: number): number => value / 2
+
+// if we originally had a value outside of an array return it that way
+// otherwise the underfined values will cause incorrect responsivness
```
Comment (callumacrae) [inline src/layout/flex-grid/src/index.tsx:23]: "underfined"

File: src/layout/flex-grid/src/index.tsx [hunk 3]
```diff
@@ -50,14 +12,23 @@ const getVerticalGutterSize = (theme: any): string | number =>
...
 
-const mediaQueryFunction = (
-  data: number | number[],
-  fn: (colValue: number, index: number) => string
-): string | string[] => {
-  if (Array.isArray(data)) {
-    return data.map(fn)
-  }
-  return fn(data, 0)
+const alwaysGetArray = (valueOrArray: any): any => [].concat(valueOrArray)
```
Comment (callumacrae) [inline src/layout/flex-grid/src/index.tsx:15]: Might be able to use generic here? https://www.typescriptlang.org/docs/handbook/generics.html

`(valueOrArray: T | T[]) T[]`

(untested - don't know if you can use it with `|` like that)

Also, you really missed an opportunity to call this `castArray` 😦

File: src/layout/flex-grid/src/index.tsx (hunk 1 above)
Comment (barneyfoxuk) [reply src/layout/flex-grid/src/index.tsx:86]: it whines without so leaving

**Pull request merged**